File: apps/party-time-backend/src/main/java/com/partytime/api/dto/login/LoginRequestDTO.java
package com.partytime.api.dto.login;

import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class LoginRequestDTO {

    @NotNull
    @NotEmpty
    private String email;

    @NotNull
    @NotEmpty
    private String password;

}

File: apps/party-time-backend-kotlin/src/main/kotlin/com/partytime/api/dto/login/LoginRequestDTO.kt
package com.partytime.api.dto.login

import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotEmpty

/**
 * Data transfer object used for login requests
 *
 * @param email Email of the account holder
 * @param password Password of the to-be-created account
 */
data class LoginRequestDTO(
    @field:NotEmpty
    @field:Email
    val email: String,
    @field:NotEmpty
    val password: String
)
